WordPress/Classes/Models/Post+RefreshStatus.swift

```diff
@@ -7,27 +7,23 @@ extension PostService {
     ///   - onCompletion: block to invoke when status update is finished.
     ///   - onError: block to invoke if any error occurs while the update is being made.
     ///
-    func refreshPostStatus(onCompletion: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
-        self.managedObjectContext.perform {
+    static func refreshStatus(with coreDataStack: CoreDataStack, onCompletion: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
```

Comment:
Do you think this would work better with a `Result` type to consolidate the two completions into one?

Anyway, that could happen in a follow up PR, if at all.

Reply:
I like having one single closure in the function signature 👍 . I've seen quite a few places where there are separate callbacks for success and failure, I'll evaluate them case by case when I change those functions since I want to keep the PRs changes focused for easier review. For this case here, these callbacks can actually be removed as they are not used by the call site. I'll make this change in a follow up PR.